docs/future/spa-mode.md

```diff
@@ -228,8 +228,7 @@ startTransition(() => {
   - Generally speaking, the way to resolve these issues is to import any browser-only libraries from `entry.client.tsx` so they don't end up in the server build
   - Otherwise, you can generally solve these by using [`React.lazy`][react-lazy] or the [`<ClientOnly>`][client-only] component from `remix-utils`
 
-- The [SPA Mode template][spa-mode-template] enables the Vite [ssr.noExternal][vite-ssr-noexternal] option by default to automatically bundle all of your dependencies during the server build to avoid most ESM/CJS issues
-  - This may slow down your build a bit — if so, you can try removing this option, or switching to a more targeted array containing the specific dependencies you wish to bundle
+- If you are running into ESM/CJS issues with your app dependencies you may need to set the Vite [ssr.external][vite-ssr-external] and/or [ssr.noExternal][vite-ssr-noexternal] options in your config to exclude or include certain dependencies in your server bundle.
```

Comment:
We may want to be more specific that `noExternal` will bundle the dependencies in your server bundle. I worry a bit that the language as it stands is going to encourage people to just flip a coin putting dependencies in `ssr.external` vs `ssr.noExternal`, whereas they really are always going to want to try `ssr.noExternal` (unless they set that to `true`, but that feels like the unhappy path at this point).

Also might be good to include a quick code snipped, in case they don't realize `ssr` is a top level key in the vite config.

Reply:
👍 

It's tricky to draw the line between how much we should explain vite stuff that's already documented on the vite pages we're linking too.  Bring too much over and our docs get bloated and potentially stale.  Bring too little over and folks who don't click through may not grok the point we're making. I'll push an update to try to make this a bit clearer though - and I agree the code snippet would be helpful

Reply:
[bf62380](https://github.com/remix-run/remix/pull/8957/commits/bf62380bd27ad37ec30af17caa1fd1aa3c5730c9)